83 develop sign up page sign up with google

SufyanCS opened the pull request:
## **Describe what does this pull request do?**
1-Implement the option for users to sign up using their Google account.
2- Modify the next.config file to display the user's profile image in the navigation bar.
3- Activate the Sign up with Google button to allow users to use this feature.
4- Upon successful login, direct the user to the events page for a seamless user experience.

SufyanCS committed:
feat(google sign up): set up Google Sign Up Functionality

Configure Google Sign Up Functionality to Allow User Sign Up with a Pop-up Window

SufyanCS committed:
feat(google sign up button): enable Google signup in ButtonAuth component

enable the "sign up with Google" functionality in the ButtonAuth component with onClick event
handler that calls the signUpwithGoogle function from the useAuth module when the Google button is
clicked

SufyanCS committed:
feat(update next.config.js file): update next.config.js for image loading

update the next.config.js file to include the lh3.googleusercontent.com domain, allowing images from
this domain to be loaded in the application. This resolves an error that was occurring when
attempting to load images from this domain

SufyanCS committed:
feat(implement google login redirect): add Google login redirect to events page

adds a redirect to the events page after the user logs in with their Google account

sulmi24 reviewed (approved):
Well Done 
keep going

iiAbady commented on lib/useAuth.js:
```diff
@@ -20,6 +24,14 @@ export async function signUp(email, password, data) {
+        await signInWithPopup(auth, googleProvider);
+        Router.push("/events");
+    } catch (err) {
+        console.error(err);
```
Very good job @SufyanCS! There is one little suggestion I have:

Instead of try catching or console erroring the error here, what you can do is returning just returning `signInWithPopup()` ,like the `signUp()` function, and then in the signin or signup page you can handle that error with the try catch while using the `AuthErrorBox` to show that error to the user.

Check the signup function in the signup page for more reference!

SufyanCS replied:
Done, Check and review the code again.

SufyanCS committed:
Merge branch 'develop' into 83-develop-sign-up-page-sign-up-with-google

SufyanCS committed:
fix(updated error handling in signupwithgoogle): improve error handling in signUpwithGoogle function

Improve error handling in signUpwithGoogle function to provide more specific error messages to
provide more informative error messages for different types of errors

SufyanCS committed:
Merge branch '83-develop-sign-up-page-sign-up-with-google' of github.com:202212-GIZ-YE-FEW/Web_Wizards_Pebble into 83-develop-sign-up-page-sign-up-with-google

SufyanCS committed:
Merge branch 'develop' into 83-develop-sign-up-page-sign-up-with-google
